setuptools_rust/build.py

```diff
@@ -349,10 +351,42 @@ def get_dylib_ext_path(self, ext, target_fname):
         assert modpath not in build_ext.ext_map
         build_ext.ext_map[modpath] = ext
         try:
-            return build_ext.get_ext_fullpath(target_fname)
+            ext_path = build_ext.get_ext_fullpath(target_fname)
         finally:
             del build_ext.ext_map[modpath]
 
+        if '.abi3.' in ext_path:
```

Comment:
The logic from here down is quite complex. Can we split it into a few smaller functions, to help with cognitive load? e.g. `is_linux_cross_compile()` , `strip_platform_and_extension()`? That might be enough.